run.py

```diff
@@ -296,16 +296,80 @@ def zip_most_outputs(directory, archive_name, promote_list):
                 os.remove(file)
 
 
-def get_subject_code(fw_client, analysis_id):
+def generate_analysis_file_label(fw_client, config_dict, extension=None, name_string=None, timestamp=False):
     """
-    Takes and input string and some gear configuration information to get the subject ID
-    :param fw_client: the flywheel client
-    :param analysis_id: analysis ID from the config/context
-    :return:
+    A reasonably robust tool for attempting to generate a unique file name for a flywheel analysis. It will try to
+    use the subject code(s) for the input files as the name, and if it cannot, it will fall back on an optionally
+    provided name_string and timestamp. Timestamp will be added whenever both name_str and subject codes are absent
+    :param fw_client: (flywheel.client.Client) An instance of the flywheel client
+    :param config_dict: (dict) a dictionary representation of config.json
+    :param extension: (str) the file extension
+    :param name_string: (str) optional additional label to add to the file label
```

Comment:
Is `name_string` to be provided via config option in the manifest?

Reply:
name_string is mostly there to allow configuration if I want to copypasta into other gears. I use it to distinguish between the report.siena(x), report.viena, and report.html